pycuda_cudf_integration/README.md

```diff
@@ -0,0 +1,113 @@
+# Using PyCuda and Cudf
+## Overview
+Sometimes, a user may wish to run custom device code to do numerical operations on a `cuDF` column. 
+
+One way to do this is through custom cython bindings which can require significant setup work as shown [here](https://github.com/rapidsai/rapids-examples/tree/main/shareable-dataframes).
+
+With `PyCUDA`,  the custom CUDA kernels can be run directly using its` sourcModule` to modify the `cuDF`  dataframe.   
+
+`PyCUDA` has limitations, especially around running host-side code, which can are noted in detail in the limitations section.  
+
+## PyCUDA Interactions
+In order to write custom cuda kernel in `pycuda` we make use of `SourceModule`. After constructing the kernel, we can retrieve the function and store it in a variable like below.
+```python
+import pycuda.autoprimaryctx
+from pycuda.compiler import SourceModule
+
+mod = SourceModule("""
+    __global__ void doublify(int64_t *a, int N)
+    {
+      int stride = blockDim.x * gridDim.x;
+      for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < N; i += stride) {
+        if (i < N) {
+          a[i] *= 2;
+        }
+      }
+    }
+    """)
+func = mod.get_function("doublify")
+```
+The interactions between `cudf` and `pycuda` is dependent upon the implementation of the `__cuda_array_interface__`, which serves as a contract to tell the world of how to convert the underlying data between array-like data structures, without requiring a copy.
+```python
+import cudf
+import cupy as cp
+
+df = cudf.DataFrame({'col': [i for i in range(200000)]})
+length = cp.int32(len(df['col']))
+
+func(df['col'], length, block=(256,1,1), grid=(4096,))
+```
+
+## PyCuda Limitations
+At its core, `pycuda` is meant for writing CUDA kernels that operate on fixed length columns. One interesting fact about `pycuda` is that it allows including external libraries, such as `thrust` or `libcudf`. However, the issue becomes that pycuda allows device-based code in external libraries such as `thrust`, but does not support running host code.
```

Comment:
`operate on fixed length columns` is this meant to be fixed width typed data? Is there a circumstance in which circumstance a data buffer would have variable length?